build-aux/autotroll.m4

```diff
@@ -223,14 +224,16 @@ AC_DEFUN([AT_WITH_QT],
      # Find Qt.
      AC_ARG_VAR([QT_PATH],
        [path to Qt binaries])
+     # HomeBrew installs into /usr/local/share/qt/libexec
+     QT_TOOL_PATH=$QT_PATH:$PATH:/usr/local/share/qt/libexec
```

Comment:
What's the reason to have a fallback directory for HomeBrew?  And what about MacPorts?  Wouldn't it be a cleaner solution to omit this completely?

Reply:
HomeBrew appears to be the most popular "package system" for macOS and it would be nice to handle that automatically (at least the default installation location). There is logic for auto-searching other common installation locations, more typically used on GNU/Linux and other Unixes.  It might be reasonable to also support MacPorts, but my impression is that it is less commonly used and less active there days,  Regardless, I don't have MacPorts installed, so I can't test it.

Reply:
FWIW I am not really a mac user and am not a macOS expert.  However, I am trying to make the [DomTerm](https://domterm.org) easier to install and work better on macOS, so I bought a refurbished MacBook Air to test things on.

Reply:
Regarding MacPorts: You are severely mistaken :-)  It is as active as HomeBrew.  In addition, and contrary to HomeBrew, it supports all MacOS systems down to Leopard whenever possible.  Currently, it has 29500 active ports, which is not too bad IMHO.  MacPorts might not be as 'hip' as HomeBrew, but it works rock-solid.  I use it for my old Mac OS Lion box.

Another argument against adding a fallback binary directory is the ability of both HomeBrew and MacPorts to install different Qt versions in parallel (which you can select, for example, by using `brew switch qt 4.8.7_2`).  On HomeBrew, there also exists the command `brew link qt` to create a link for qmake into `/usr/local/bin` – MacPorts has something similar.

So: please remove the fallback directory.

Reply:
I checked in an update.  I left in the `QT_TOOL_PATH` as a helpful refactoring.

One idea: If QT_PATH is empty, default it to a copy of `$PATH` but with each `/bin` replaced by `/libexec`. What do you think of that idea? Possibly optimized to only add existing directories, but that gets more complicated, and is probably not worth it.

Reply:
Thanks for the update, which looks good.

Regarding an empty `QT_PATH`, I'm undecided.  Honestly, I would prefer if we rely on the developer setting up the correct directory in `PATH` to the `qmake` binary he or she wants to use.  In many cases, guessing defaults is problematic.  Maybe you can investigate what other projects like 'cmake' or 'meson' do?  I think they have to solve exactly the same problem for Qt, right?

Reply:
CMake seems to be the preferred build tool for Qt projects these days.
See [https://doc.qt.io/qt-6/cmake-manual.html](https://doc.qt.io/qt-6/cmake-manual.html) and [Getting started with CMake](https://doc.qt.io/qt-6/cmake-get-started.html).  You would use a line like:

    find_package(Qt6 REQUIRED COMPONENTS Widgets)

  I don't really know how [find_package](https://cmake.org/cmake/help/latest/command/find_package.html) works, but it seems like an enhanced `PKG_CONFIG`.

Reply:
`find_package()` is basically "PKG_CONFIG, if PKG_CONFIG was written in a turing-complete scripting language so that only cmake can use it at all; also it allows providing arbitrary macros".

So it's more like m4 macros than pkg-config, which makes it very problematic to use find_package in autotools or Meson, for the same reason it's very problematic to use m4 macros in Meson or cmake.

pkg-config is supposed to be the common glue that connects build systems together for discoverability -- m4 macros, cmake find_package macros, and meson modules should all be using pkg-config under the hood for reading information out of a software installation such as Qt.

Meson does support a custom overridden logic for scraping qmake information: https://github.com/mesonbuild/meson/blob/master/mesonbuild/dependencies/qt.py

Reply:
Indeed, using `pkg-config` would be beneficial.  Volunteers to update `autotroll.m4` accordingly are highly welcomed :-)